src/test/java/spoon/test/textBlocks/TextBlockTest.java

```diff
@@ -118,4 +119,37 @@ public void testTextBlockCreation(){
 				c.toString()
 		);
 	}
+
+	@Test
+	@ExtendWith(LineSeperatorExtension.class)
+	public void testTextBlockEscapes(){
+		//contract: text-blocks should retain escape sequences in code
+		Launcher launcher = setUpTest();
+		launcher.getEnvironment().setAutoImports(true);
+		
+		CtClass<?> allstmt = (CtClass<?>) launcher.getFactory().Type().get("textBlock.TextBlockTestClass");
+		CtMethod<?> m5 =  allstmt.getMethod("m5");
+
+		CtStatement stmt5 = m5.getBody().getStatement(0);
+		assertTrue(stmt5.getValueByRole(CtRole.ASSIGNMENT) instanceof CtTextBlock);
+
+		// test text block value
+		CtTextBlock l1 = (CtTextBlock) stmt5.getValueByRole(CtRole.ASSIGNMENT);
+		assertEquals("no-break space: \\00a0\n" +
+						"newline:        \\n\n" +
```

Comment:
Using `replace("\\", "\\\\")` would only escape the backslashes and not `\n` because the latter is treated as a single character and not a combination of `\` + `n`. For example, if you run `replace("\\", "\\\\")` on `\\\n`, it would return `\\\\\n`. Notice that `\n` is not escaped. I think it would be better if we write a generic function to escape all characters or use one which already exists. I have something like this in mind:
```java
public static String escape(String s){
  return s.replace("\\", "\\\\")
          .replace("\t", "\\t")
          .replace("\b", "\\b")
          .replace("\n", "\\n")
          .replace("\r", "\\r")
          .replace("\f", "\\f")
          .replace("\'", "\\'")
          .replace("\"", "\\\"");
}
```

Reply:
As long as in the source code it's written as "\n" it works because at that time it's still two characters. It is not supposed to escape the newline at the end of a line in a textblock. The unit test for this passes and the original problem I had with spoon is also solved by this patch, so I am quote sure it works.

Reply:
> As long as in the source code it's written as "\n" it works because at that time it's still two characters.

Okay. So do we just want to escape backslashes when we put the literal inside a text block?

Reply:
Yes, "normal" String literals are handled in LiteralHelper.getLiteralToken(), which in turn calls LiteralHelper.getStringLiteral(). But if we use that one, all newlines, tabs etc. are replaced by their respective escape sequences and all textblocks would be printed on a single line.

Reply:
Okay. Your patch looks good to me in that case.